src/m365/commands/login.ts

```diff
@@ -239,7 +243,14 @@ class LoginCommand extends Command {
       }
     }
     finally {
-      logout();
+      if (!args.options.ensure ||
```

Comment:
We're missing the comparison of the authType. If I'm signed in using device code and then run `m365 login --ensure --authType browser` I'm not getting a login flow. We showed this example in the original issue: https://github.com/pnp/cli-microsoft365/issues/5217#issue-1803533744. Let's add it